designs/extensible-cli-and-scaffolding-plugins-phase-2.md

```diff
@@ -0,0 +1,203 @@
+# Extensible CLI and Scaffolding Plugins - Phase 2
+
+## Overview
+
+* Plugin Phase 1.5 was desgined and implemented to achieve chaining of plugins. The main goal of Phase 2 plugins is Kubebuilder discovering and using external plugins, also referred as out-of-tree plugins. So Phase 2 achieves both chaining of plugins and discovery of external plugins/source code which is not intented to be shipped with the binaries. For e.g., Kubebuilder being able to discover the helm plugin and use that without having to build the plugin with the binary. These external plugins can be implemented in any language.
```

Comment:
```suggestion

UPlugin [Phase 1.5](thttps://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/extensible-cli-and-scaffolding-plugins-phase-1-5.md) was designed and implemented to allow chaining plugins. 

The purpose of Phase 2 plugins is to allow discovering and using external plugins, also referred to as out-of-tree plugins. So Phase 2 achieves both chainings of plugins and discovery of external plugins/source code which is not intended to be shipped with the CLI binary. By achieving this goal, we could externalize the optional [declarative](https://github.com/kubernetes-sigs/kubebuilder/tree/master/pkg/plugins/golang/declarative/v1) which means that the CLI would still able to use it, however, its code source no longer would require to be inside of the Kubebuilder repository as built with the CLI release. 
```

We have not Helm plugin in KB. In this way, I would suggest we stick to what we can find here.

Reply:
I used helm plugin because this EP is about external plugins that can be written in any language. But I like the idea of using the declarative plugin example and mentioning about externalizing it with Phase 2 implementation. Fixed that.